plugins/Alias/config.py

```diff
@@ -43,5 +43,7 @@ def configure(advanced):
 Alias = conf.registerPlugin('Alias')
 conf.registerGroup(Alias, 'aliases')
 conf.registerGroup(Alias, 'escapedaliases')
+conf.registerGlobalValue(Alias, 'validName',
+    registry.String(r'^[^\x00-\x20]+$', _("""Regex which alias names must match in order to be valid""")))
```

Comment:
Strings written in the config file are restricted by supybot.registry, so:
":", ".", and "\" are escaped ("." is escaped by Alias too).
Alias names cannot contain whitespace (which includes newlines), since len(name.split()) must equal 1 otherwise the registry will refuse to accept the name.